### PR TITLE
Add .travis.yml

### PR DESCRIPTION
**Pull request by kjmahalingam**
Adds a `.travis.yml` file which tests the `shr_spec` against the most recent master of `shr-cli`.

It checks to see that there are no errors in running the `shr-cli` against the `shr_spec` and passes the build in this case. Otherwise, the build fails.

### CODE REVIEW

**Comment by kjmahalingam** [comment]
I have implemented the caching of `node_modules`, as requested above. I have looked into cachcing `shr-cli`, but am having difficulty finding a way to do so for a repository that is cloned during operation. I could perhaps lower the git clone depth to 1, to speed up the builds a bit. Or if you have additional suggestions, I am happy to look more into them.

**Comment by okeefm** [comment]
That might be worth doing, to try to make it less painful to clone. I'm ok with not caching, assuming we're able to do that.

**Comment by kjmahalingam** [comment]
I have lowered the git clone depth to 1, as discussed above.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+language: node_js
+node_js:
+  - '8'
+cache:
```
Comment (okeefm) [inline .travis.yml]: Is this how one caches `yarn` dependencies? It looks from the docs like you do something like: `cache: yarn`

Comment (kjmahalingam) [reply .travis.yml]: You're right. It's supposed to be that way. I just used `npm` instead of `yarn` by accident!

**Pull request merged**